interfaces/cython/cantera/yaml2ck.py

```diff
@@ -0,0 +1,714 @@
+
+# Conversion from 1 debye to coulomb-meters
+DEBEYE_CONVERSION = 3.33564e-30
+
+def reorder_reaction_equation(solution, reaction):
+    # Split Reaction Equation
+    rxn_eqn = reaction.equation
+    for reaction_direction in [' <=> ', ' <= ', ' => ']:
+        if reaction_direction in rxn_eqn:
+            break
+    for third_body in [' (+M)', ' + M', '']: # search eqn for third body
+        if third_body in rxn_eqn:            # if reaches '', doesn't exist
+            break
+
+    # Sort and apply to reaction equation
+    reaction_txt = []
+    reaction_split = {'reactants': reaction.reactants, 
+                      'products': reaction.products}
+    for n, (reaction_side, species) in enumerate(reaction_split.items()):
+        species_weights = []
+        for key in species.keys():
+            index = solution.species_index(key)
+            species_weights.append(solution.molecular_weights[index])
+        
+        # Append coefficient to species
+        species_list = []
+        for species_text, coef in species.items():
+            if coef == 1.0:
+                species_list.append(species_text)
+            else:
+                species_list.append(f'{coef:.0f} {species_text}')
+                
+        species = species_list
+        
+        # Reorder species based on molecular weights
+        species = [x for y, x in sorted(zip(species_weights, species))][::-1]
+        reaction_txt.append(' + '.join(species) + third_body)
+    
+    reaction_txt = reaction_direction.join(reaction_txt)
+    
+    return reaction_txt
+   
+
+def eformat(f, precision=7, exp_digits=3):
+    s = f"{f: .{precision}e}"
+    if s == ' inf' or s == '-inf':
+        return s
+    else:
+        mantissa, exp = s.split('e')
+        exp_digits += 1 # +1 due to sign
+        return f"{mantissa}E{int(exp):+0{exp_digits}}" 
+  
+ 
+def build_arrhenius(rate, reaction_order, reaction_type):
+    """Builds Arrhenius coefficient string based on reaction type.
+    Parameters
+    ----------
+    rate : cantera.Arrhenius
+        Arrhenius-form reaction rate coefficient
+    reaction_order : int or float
+        Order of reaction (sum of reactant stoichiometric coefficients)
+    reaction_type : {cantera.ElementaryReaction, cantera.ThreeBodyReaction, cantera.PlogReaction}
+        Type of reaction
+    Returns
+    -------
+    str
+        String with Arrhenius coefficients
+    """
+    if reaction_type in [ct.ElementaryReaction, ct.PlogReaction]:
+        pre_exponential_factor = rate.pre_exponential_factor * 1e3**(reaction_order - 1)
+
+    elif reaction_type == ct.ThreeBodyReaction:
+        pre_exponential_factor = rate.pre_exponential_factor * 1e3**reaction_order
+
+    elif reaction_type in [ct.FalloffReaction, ct.ChemicallyActivatedReaction]:
+        raise ValueError('Function does not support falloff or chemically activated reactions')
+    else:
+        raise NotImplementedError('Reaction type not supported: ', reaction_type)
+    
```

Comment:
In the (4) cases on `NotImplementedError`, would it make sense to have the blocks of thermo or reaction data written to stdout or to a text file so that an end-user knows exactly what needs to be converted manually to build out a CK input deck?
I don't think doing any more than that for reactions would be appropriate since we don't want to endorse an approximate mapping

Reply:
I've altered the error messages to include what is triggering the errors.